src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -1410,6 +1416,84 @@ class DevTask extends AbstractFeatureTask {
         }
     }
 
+    private List<ProjectModule> getProjectModules() {
+        List<ProjectModule> upstreamProjects = new ArrayList<ProjectModule>();
+
+        HashSet<Project> allProjects = new HashSet<>()
+        allProjects.addAll(DevTaskHelper.getAllUpstreamProjects(project))
+        for (Project dependencyProject : allProjects) {
+            // TODO get compiler options for upstream project
+            // JavaCompilerOptions upstreamCompilerOptions = getMavenCompilerOptions(p);
+            JavaCompilerOptions upstreamCompilerOptions = new JavaCompilerOptions();
+            SourceSet mainSourceSet = dependencyProject.sourceSets.main;
+            SourceSet testSourceSet = dependencyProject.sourceSets.test;
+            DefaultFilePropertyFactory.DefaultDirectoryVar outputDirectory = mainSourceSet.java.classesDirectory;
+            DefaultFilePropertyFactory.DefaultDirectoryVar testOutputDirectory = testSourceSet.java.classesDirectory;
+            Set<String> compileArtifacts = new HashSet<String>();
+            Set<String> testArtifacts = new HashSet<String>();
+            File upstreamSourceDir = new File(mainSourceSet.java.sourceDirectories.asPath)
+            File upstreamOutputDir = outputDirectory.asFile.get();
+            File upstreamTestSourceDir = new File(testSourceSet.java.sourceDirectories.asPath);
```

Comment:
You are referencing a lot of different configurations in the preceding lines....could some of them not exist and end up throwing an exception?

Reply:
All these will be always present for any java project. We are using similarly in LooseEarApplication